index.js

```diff
@@ -71,21 +64,56 @@ app.get('/minimal', async (req, res) => {
 });
 
 app.get('/oauth', (req, res) => {
-  const authorizationUri = oauthClient.code.getUri()
+  console.log("oauth: begin initiation");
+  // console.log("Request Headers: ", req.headers);
+
+  const authEndpoint = process.env.OAUTH_AUTH_URI;
+  const verifier = base64url(crypto.pseudoRandomBytes(32));
+  req.session.verifier = verifier;
+  const challenge = base64url(crypto.createHash('sha256').update(verifier).digest());
+
+  const queryParams = new URLSearchParams({
+    response_type: 'code',
+    client_id: process.env.OAUTH_CLIENT_ID,
+    redirect_uri: process.env.OAUTH_REDIRECT_URI,
+    code_challenge: challenge,
+    code_challenge_method: 'S256'
+  })
 
-  res.redirect(authorizationUri)
+  const authorizationUri = `${authEndpoint}?${queryParams}`
+
+  res.redirect(authorizationUri);
 });
 
 app.get('/callback', async (req, res) => {
-  const credentials = await oauthClient.code.getToken(req.originalUrl);
-
-  res.render('index', {
-    host,
-    token: credentials.accessToken,
-    destination: process.env.DEFAULT_DESTINATION,
-    firebaseConfig: FIREBASE_CONFIG,
+  console.log("oauth: process callback");
+  // console.log("Request Headers: ", req.headers);
+
+  const params = new URLSearchParams();
+  params.append('client_id', process.env.OAUTH_CLIENT_ID);
+  params.append('grant_type', 'authorization_code');
+  params.append('code', req.query.code);
+  params.append('redirect_uri', process.env.OAUTH_REDIRECT_URI);
+  params.append('code_verifier', req.session.verifier);
+
+  const response = await fetch(process.env.OAUTH_TOKEN_URI, {
```

Comment:
We have `axios` installed as well. I think for consistency, we should stick to one. Maybe we can remove the axios.

Reply:
Yeah, seeing that auth.js relies on fetch made we wonder why we use axios still, I can convert the other method and remove axios if we think removing it is fine.